src/core/query.js

```diff
@@ -664,7 +664,8 @@ export class Query {
    * @returns {Query} The query.
    */
   addFilter(field, condition, values) {
-    if (isDefined(condition) && isDefined(values)) {
+    if (isDefined(condition)
+        && (isDefined(values) || condition === '$ne')) {
```

Comment:
It seems to me that each separate filter method should validate the values for itself. If the check is in here, this method needs to know what values are valid for which operator. If we need to allow this for another method, we'd need another "or" statement. Just something to think about, I won't request changes - it's up to you.

Reply:
Yes this is something I thought about. For now this had the least impact to fix the bug. I will look over it again to see if there is a better way but at first glance I did not see another option.